OpenCL_C.txt

```diff
@@ -4457,31 +4502,44 @@ all arguments and the return type, unless otherwise specified.
 
   float__n__ **lgamma_r**(float__n__ _x_, {global} int__n__ *_signp_) +
   float **lgamma_r**(float _x_, {global} int *_signp_) +
-  double__n__ **lgamma_r**(double__n__ _x_, {global} int__n__ *_signp_) +
-  double **lgamma_r**(double _x_, {global} int *_signp_) +
 
   float__n__ **lgamma_r**(float__n__ _x_, {local} int__n__ *_signp_) +
   float **lgamma_r**(float _x_, {local} int *_signp_) +
-  double__n__ **lgamma_r**(double__n__ _x_, {local} int__n__ *_signp_) +
-  double **lgamma_r**(double _x_, {local} int *_signp_) +
 
   float__n__ **lgamma_r**(float__n__ _x_, {private} int__n__ *_signp_) +
   float **lgamma_r**(float _x_, {private} int *_signp_) +
+
+  For OpenCL C 2.0 or with the `+__opencl_c_generic_address_space+`
+  feature macro: +
+
+  float__n__ **lgamma_r**(float__n__ _x_, int__n__ *_signp_) +
+  float **lgamma_r**(float _x_, int *_signp_)
+    | Log gamma function.
+      Returns the natural logarithm of the absolute value of the gamma
+      function.
+      The sign of the gamma function is returned in the _signp_ argument of
+      *lgamma_r*.
+| double__n__ **lgamma_r**(double__n__ _x_, {global} int__n__ *_signp_) +
```

Comment:
Does these `double lgamma_r`  definitions need  a `Requires support for doubles, e.g. with the +__opencl_c_fp64+ feature macro. ` comment too

Reply:
So it already has this.

I'm a bit inconsistent in how I do this, but not without reason.   Rows were some built-ins use doubles get a comment next to the applicable functions in the left column, rows were all the built-ins use doubles get a comment in the right column.   So here it's in the right column, but only because I split `lgamma_r` into two rows here.  It makes more sense in the rendered form.